web/src/user_profile.ts

```diff
@@ -819,14 +931,23 @@ function get_human_profile_data(fields_user_pills) {
     return new_profile_data;
 }
 
-function get_current_values($edit_form) {
-    const current_values = dialog_widget.get_current_values(
+function get_current_values(
+    $edit_form: JQuery,
+): Record<string, unknown> & {user_id?: string | number | undefined} {
+    const raw_current_values = dialog_widget.get_current_values(
         $edit_form.find("input, select, textarea, button, .pill-container"),
     );
+    const schema = z.intersection(
+        z.object({
+            user_id: z.optional(z.union([z.string(), z.number()])),
+        }),
+        z.record(z.string(), z.unknown()),
+    );
```

Comment:
I wasn't super confident about user_id being `string` type only so I've kept it as a union of both `string` and `number`

Reply:
I will try to investigate it further, though we can give the schema a better name

Reply:
Oh yes, thanks! I named that temporarily and forgot to rename it.

Reply:
Why does this parsing do? It's pretty hard to read htis block of code.

Reply:
It parses `Record<string | unknown>` to check whether the key `user_id` exists or not.

Reply:
It might make sense to convert to storing an integer in a follow-up PR, but I've not looked at the details of whether that would simplify things.